.travis/set_python_version.sh

```diff
@@ -14,16 +15,20 @@ if [[ -n $PYTHON_SUFFIX ]]; then
     chmod +x $PYTHON_WRAPPER
 
     export PATH=~/pythonbin:$PATH
-    PYVER=$(python -c 'import sys; print("%d.%d." % (sys.version_info.major, sys.version_info.minor))') || STATUS_CODE=1
+    PYVER=$(python -c 'import sys; print("%d.%d." % (sys.version_info.major, sys.version_info.minor))')
     echo "Check python version"
     if [[ $PYVER != "$PYTHON_SUFFIX."* ]]; then
         echo "Error: Python binary suffix is $PYTHON_SUFFIX, but version reported is $PYVER"
         echo "Set up Python version: FAIL"
-        return 1
+        PYTHON_ERR=1
     fi
 else
     echo "Error: No Python version selected"
-    return 1
+    PYTHON_ERR=1
 fi
 
-return 0
+# If we had a problem above, run false to set the exit code so the
```

Comment:
What exactly was going wrong here, vs using `return`?

Reply:
This was me seeing a few different errors in Travis, twiddling a few things to check and cleaning up.

I'll re-instate the returns, as they should work and it's a bit cleaner (though I'm not keen on a. source'd scripts returning b. source'd scripts using exit codes)

Reply:
Thanks! Yeah I guess ideally they should be in a function.